utilities/xmlrpcpp/include/xmlrpcpp/XmlRpcClient.h

```diff
@@ -118,6 +118,9 @@ namespace XmlRpc {
     // each thread should have its own client.
     bool _executing;
 
+    // True if the server supports HTTP1.1
+    bool _server_http_1_1;
```

Comment:
Can this be `private`?

Reply:
> but this breaks ABI.

our bad, that should be kept...sorry.

Reply:
I knew it breaks ABI, but it's clear.
I tried to use `_connectionState`, but some source code might be as 

```cpp
const int SERVER_HTTP_DISCONNECT = 0x10000;
const int CLIENT_CONNECT_STATE_MASK = 0xFFFF;

check header to set _connectionState:
  _connectionState |= SERVER_HTTP_DISCONNECT;

set:
_connectionState = READ_HEADER;
  -->
_connectionState |= READ_HEADER;


if (_connectionState == WRITE_REQUEST)
   -->
if (_connectionState & CLIENT_CONNECT_STATE_MASK == WRITE_REQUEST)

check if SERVER_HTTP_DISCONNECT:
if (_connectionState & SERVER_HTTP_DISCONNECT)
```

do you think it's acceptable?

Reply:
Do you have any other suggestions? I'd like to hear that.

Reply:
I'll make use of the `_header` member variable, make it [not clean](https://github.com/ros/ros_comm/blob/63c9bbb5c2480a232530cfa13308a22d727f20b8/utilities/xmlrpcpp/src/XmlRpcClient.cpp#L457) after `XmlRpcClient::readHeader`, and then check it [here](https://github.com/ros/ros_comm/blob/63c9bbb5c2480a232530cfa13308a22d727f20b8/utilities/xmlrpcpp/src/XmlRpcClient.cpp#L132) and clean it.